bfrec

```diff
@@ -325,6 +329,52 @@ EOF
     return 0
 }
 
+has_debug_bfb()
+{
+    # A debug image consists of a regular BFB file which
+    # includes an authorization certificate (a.k.a. AuthCert).
+    # The debug image is expected to be installed in Secure
+    # LifeCycle state.
+    #
+    # Implement a dumb check to verify whether the
+    # authorization certificate is present within the
+    # BFB. It is a basic indication that the BFB is
+    # intended for debug.
+
+    in_file=$1
+
+    # Read Arm lifecycle, debug image is ignored in lifecycle state non-secure.
+    lc=$(mlxbf-bootctl | grep lifecycle | cut -f2 -d':' | tr -d ' ')
+    if [ "$lc" != "Secured(development)" ] && [ "$lc" != "GASecured" ]; then
+        return 0
+    fi
+
+    authcert_cn="Trusted Debug BF Root Authorization Certificate"
+    # Add dummy 'tr' command, because 'grep' might fail and cause the script
+    # to exit as '-e' is set. 
+    authcert="$(strings "$in_file" | grep -m 1 "$authcert_cn" | tr -d ' ')"
+    if [ -z "$authcert" ]; then
+        return 0
+    fi
+
+    # If we reach here, it means that a debug image is
+    # going to be installed, thus request user confirmation.
+    cat <<EOF
+
+    ***********************************************************************
+    ***                                                                 ***
+    ***   You are about to install a debug image on Secured platform.   ***
+    ***                                                                 ***
+    ***********************************************************************
+EOF
+    read -p "Continue (Y|n)[n]: " -n 1 -r
```

Comment:
Apologies if I am missing this here , is there a string output before the "Continue (Y|n)[n:] " and if so just wondering what it says?

Reply:
Yes, "You are about to install a debug image on Secured platform" (as shown at lines 364-368) is displayed prior to "Continue (Y|n)[n]:". The intention is to draw attention. Why? because usually this command doesn't require user intervention.